src/GrumPHP/Locator/ChangedFiles.php

```diff
@@ -41,10 +41,23 @@ public function __construct(Repository $repository, Filesystem $filesystem)
     public function locateFromGitRepository()
     {
         $diff = $this->repository->getWorkingCopy()->getDiffStaged();
-
         return $this->parseFilesFromDiff($diff);
     }
+    
+    /**
+     * @return FilesCollection
+     */
+    public function locateFromGitPushedRepository()
+    {
+        $diff = explode("\n", \Gitonomy\Git\Repository::run('diff origin/master..HEAD --name-only --oneline'));
```

Comment:
for us is not enough because we need to know what is the remote branch, a repo can have different remote so I think that we needto ask to them only if use `grumphp pre-push` because on hook we already know that info.